homeassistant/components/qld_bushfire/geo_location.py

```diff
@@ -0,0 +1,230 @@
+"""Support for Queensland Bushfire Alert Feeds."""
+from datetime import timedelta
+import logging
+from typing import Optional
+
+import voluptuous as vol
+
+from homeassistant.components.geo_location import (
+    PLATFORM_SCHEMA, GeolocationEvent)
+from homeassistant.const import (
+    ATTR_ATTRIBUTION, CONF_LATITUDE, CONF_LONGITUDE,
+    CONF_RADIUS, CONF_SCAN_INTERVAL, EVENT_HOMEASSISTANT_START)
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect, dispatcher_send)
+from homeassistant.helpers.event import track_time_interval
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_CATEGORY = 'category'
+ATTR_EXTERNAL_ID = 'external_id'
+ATTR_PUBLICATION_DATE = 'publication_date'
+ATTR_STATUS = 'status'
+ATTR_UPDATED_DATE = 'updated_date'
+
+CONF_CATEGORIES = 'categories'
+
+DEFAULT_RADIUS_IN_KM = 20.0
+DEFAULT_UNIT_OF_MEASUREMENT = 'km'
+
+SCAN_INTERVAL = timedelta(minutes=5)
+
+SIGNAL_DELETE_ENTITY = 'qld_bushfire_delete_{}'
+SIGNAL_UPDATE_ENTITY = 'qld_bushfire_update_{}'
+
+SOURCE = 'qld_bushfire'
+
+VALID_CATEGORIES = [
+    'Emergency Warning',
+    'Watch and Act',
+    'Advice',
+    'Notification',
+    'Information',
+]
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_LATITUDE): cv.latitude,
+    vol.Optional(CONF_LONGITUDE): cv.longitude,
+    vol.Optional(CONF_RADIUS, default=DEFAULT_RADIUS_IN_KM): vol.Coerce(float),
+    vol.Optional(CONF_CATEGORIES, default=[]):
+        vol.All(cv.ensure_list, [vol.In(VALID_CATEGORIES)]),
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Queensland Bushfire Alert Feed platform."""
+    scan_interval = config.get(CONF_SCAN_INTERVAL, SCAN_INTERVAL)
+    coordinates = (config.get(CONF_LATITUDE, hass.config.latitude),
+                   config.get(CONF_LONGITUDE, hass.config.longitude))
+    radius_in_km = config[CONF_RADIUS]
+    categories = config[CONF_CATEGORIES]
+    # Initialize the entity manager.
+    feed = QldBushfireFeedEntityManager(
+        hass, add_entities, scan_interval, coordinates, radius_in_km,
+        categories)
+
+    def start_feed_manager(event):
+        """Start feed manager."""
+        feed.startup()
+
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, start_feed_manager)
+
+
+class QldBushfireFeedEntityManager:
+    """Feed Entity Manager for Qld Bushfire Alert GeoRSS feed."""
+
+    def __init__(self, hass, add_entities, scan_interval, coordinates,
+                 radius_in_km, categories):
+        """Initialize the Feed Entity Manager."""
+        from georss_qld_bushfire_alert_client import \
```

Comment:
This can be moved to the top of the module after introducing manifest.json.

Reply:
OK. I moved the import up and also ran `isort` while I was there.